2-api/server.js

```diff
@@ -0,0 +1,226 @@
+const express = require("express");
+const app = express();
+const { Pool } = require("pg");
+const secrets = require("./_secrets");
+
+app.use(express.json());
+
+// over the top? Yes.
+const pool = new Pool({
+  host: secrets.host,
+  port: secrets.port,
+  user: secrets.user,
+  password: secrets.password,
+  database: secrets.database,
+});
+
+// GET "/" : signposting
+app.get("/", (req, res) => {
+  res.json({ message: "Welcome to the database API! Please do something" });
+});
+
+// GET "/customers" : serve all customers in the database
+app.get("/customers", async (req, res) => {
+  const { rows } = await pool.query(
+    "SELECT name, address, city, country FROM customers;"
+  );
+
+  res.json(rows);
+});
+
+// GET "/customers/${customerId}/orders" : serves all order info from a specific customer
+app.get("/customers/:customerId/orders", async (req, res) => {
+  const { customerId } = req.params;
+  const query = `SELECT orders.customer_id, orders.order_reference, orders.order_date, products.product_name, product_availability.unit_price, suppliers.supplier_name, order_items.quantity
+  FROM orders 
+  INNER JOIN order_items ON order_items.order_id = orders.id
+  INNER JOIN product_availability ON product_availability.prod_id = order_items.product_id
+  INNER JOIN suppliers ON suppliers.id = product_availability.supp_id
+  INNER JOIN products ON products.id = product_availability.prod_id
+  WHERE orders.customer_id = $1;`;
+
+  const { rows } = await pool.query(query, [customerId]);
+
+  rows.length === 0
+    ? res.status(400).json({
+        success: false,
+        message: `No customers with ID of ${customerId} was found`,
+      })
+    : res.json(rows);
+});
+
+// GET "/customers/${customerId}" : serve customer info based on id
+app.get("/customers/:customerId", async (req, res) => {
+  const { customerId } = req.params;
+  const query =
+    "SELECT name, address, city, country FROM customers WHERE id = $1";
+
+  const { rows } = await pool.query(query, [customerId]);
+
+  rows.length
+    ? res.json(rows)
+    : res.status(400).json({
+        success: false,
+        message: `No customer with id of ${customerId} was found!`,
+      });
+});
+
+// POST "/customers" : add a new customer to the database. Name is the only thing that can't be null
+app.post("/customers", async (req, res) => {
+  const { name, address, city, country } = req.body;
+  const query =
+    "INSERT INTO customers (name, address, city, country) VALUES ($1, $2, $3, $4) RETURNING *;";
+
+  if (!name)
```

Comment:
found out that using !name.length avoids certain errors, what errors you might ask, who knows lol